builder/dockerfile/instructions/parse.go

```diff
@@ -271,16 +272,27 @@ func parseFrom(req parseRequest) (*Stage, error) {
 		return nil, err
 	}
 
+	flPlatform := req.flags.AddString("platform", "")
 	if err := req.flags.Parse(); err != nil {
 		return nil, err
 	}
+	specPlatform := system.ParsePlatform(flPlatform.Value)
+	if err := system.ValidatePlatform(specPlatform); err != nil {
+		return nil, fmt.Errorf("invalid platform %q on FROM", flPlatform.Value)
+	}
+	if specPlatform.OS != "" && !system.IsOSSupported(specPlatform.OS) {
```

Comment:
This shouldn't be in the parser package. The point of this package is to just convert from strings into typed structures. Validation should be based on the structures parser returns.

Reply:
@tonistiigi Fixed, update pushed.